File: app/src/main/java/eu/warble/pjapp/ui/main/student_info/fees/StudentFeesFragment.java
package eu.warble.pjapp.ui.main.student_info.fees;


import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.squareup.leakcanary.RefWatcher;

import eu.warble.pjapp.Application;
import eu.warble.pjapp.R;
import eu.warble.pjapp.ui.base.BaseFragment;
import eu.warble.pjapp.ui.list.ListActivity;

public class StudentFeesFragment extends BaseFragment<StudentFeesPresenter> {

    private TextView balance, balanceReceivables, balancePayments, balanceWithdrawals;
    private TextView paymentsDate1, paymentsDate1Amount, paymentsDate2, paymentsDate2Amount;
    private TextView receivables_item1, receivables_item1_date, receivables_item1_amount, receivables_item2,
             receivables_item2_date, receivables_item2_amount;
    private TextView account;

    public static StudentFeesFragment newInstance(){
        StudentFeesFragment fragment = new StudentFeesFragment();
        return fragment;
    }

    @Override
    protected StudentFeesPresenter createPresenter() {
        return new StudentFeesPresenter(this);
    }

    @Override
    protected View initView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_student_fees, container, false);
        initViews(view);
        return view;
    }

    private void initViews(View parent) {
        //balance card
        balanceReceivables = parent.findViewById(R.id.balance_receivables);
        balancePayments = parent.findViewById(R.id.balance_payments);
        balanceWithdrawals = parent.findViewById(R.id.balance_withdrawals);
        balance = parent.findViewById(R.id.balance);
        //payments card
        paymentsDate1 = parent.findViewById(R.id.payments_date1);
        paymentsDate1Amount = parent.findViewById(R.id.payments_date1_amount);
        paymentsDate2 = parent.findViewById(R.id.payments_date2);
        paymentsDate2Amount = parent.findViewById(R.id.payments_date2_amount);
        parent.findViewById(R.id.btn_payments_history).setOnClickListener(l -> startListActivity("payment"));
        //receivables card
        receivables_item1 = parent.findViewById(R.id.receivables_item1);
        receivables_item1_date = parent.findViewById(R.id.receivables_item1_date);
        receivables_item1_amount = parent.findViewById(R.id.receivables_item1_amount);
        receivables_item2 = parent.findViewById(R.id.receivables_item2);
        receivables_item2_date = parent.findViewById(R.id.receivables_item2_date);
        receivables_item2_amount = parent.findViewById(R.id.receivables_item2_amount);
        parent.findViewById(R.id.btn_receivables_history).setOnClickListener(l -> startListActivity("receivable"));
        //account card
        account = parent.findViewById(R.id.account);
    }

    private void startListActivity(String key){
        Intent intent = new Intent(getActivity(), ListActivity.class);
        intent.putExtra("key", key);
        startActivity(intent);
    }

    public void setBalance(String balance) {
        this.balance.setText(balance);
    }

    public void setBalanceReceivables(String balanceReceivables) {
        this.balanceReceivables.setText(balanceReceivables);
    }

    public void setBalancePayments(String balancePayments) {
        this.balancePayments.setText(balancePayments);
    }

    public void setBalanceWithdrawals(String balanceWithdrawals) {
        this.balanceWithdrawals.setText(balanceWithdrawals);
    }

    public void setPaymentsDate1(String paymentsDate1) {
        this.paymentsDate1.setText(paymentsDate1);
    }

    public void setPaymentsDate1Amount(String paymentsDate1Amount) {
        this.paymentsDate1Amount.setText(paymentsDate1Amount);
    }

    public void setPaymentsDate2(String paymentsDate2) {
        this.paymentsDate2.setText(paymentsDate2);
    }

    public void setPaymentsDate2Amount(String paymentsDate2Amount) {
        this.paymentsDate2Amount.setText(paymentsDate2Amount);
    }

    public void setReceivables_item1(String receivables_item1) {
        this.receivables_item1.setText(receivables_item1);
    }

    public void setReceivables_item1_date(String receivables_item1_date) {
        this.receivables_item1_date.setText(receivables_item1_date);
    }

    public void setReceivables_item1_amount(String receivables_item1_amount) {
        this.receivables_item1_amount.setText(receivables_item1_amount);
    }

    public void setReceivables_item2(String receivables_item2) {
        this.receivables_item2.setText(receivables_item2);
    }

    public void setReceivables_item2_date(String receivables_item2_date) {
        this.receivables_item2_date.setText(receivables_item2_date);
    }

    public void setReceivables_item2_amount(String receivables_item2_amount) {
        this.receivables_item2_amount.setText(receivables_item2_amount);
    }

    public void setAccount(String account) {
        this.account.setText(account);
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/mvp/studentinfo/fees/StudentFeesFragment.kt
package eu.warble.pjappkotlin.mvp.studentinfo.fees

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import eu.warble.pjappkotlin.R
import eu.warble.pjappkotlin.mvp.ApplicationNavigator
import eu.warble.pjappkotlin.mvp.BaseActivity
import eu.warble.pjappkotlin.mvp.BaseFragment
import eu.warble.pjappkotlin.utils.Constants
import eu.warble.pjappkotlin.utils.Injection
import kotlinx.android.synthetic.main.card_account.account
import kotlinx.android.synthetic.main.card_balance.balance
import kotlinx.android.synthetic.main.card_balance.balance_payments
import kotlinx.android.synthetic.main.card_balance.balance_receivables
import kotlinx.android.synthetic.main.card_balance.balance_withdrawals
import kotlinx.android.synthetic.main.card_payments.payments_date1
import kotlinx.android.synthetic.main.card_payments.payments_date1_amount
import kotlinx.android.synthetic.main.card_payments.payments_date2
import kotlinx.android.synthetic.main.card_payments.payments_date2_amount
import kotlinx.android.synthetic.main.card_payments.view.btn_payments_history
import kotlinx.android.synthetic.main.card_receivables.receivables_item1
import kotlinx.android.synthetic.main.card_receivables.receivables_item1_amount
import kotlinx.android.synthetic.main.card_receivables.receivables_item1_date
import kotlinx.android.synthetic.main.card_receivables.receivables_item2
import kotlinx.android.synthetic.main.card_receivables.receivables_item2_amount
import kotlinx.android.synthetic.main.card_receivables.receivables_item2_date
import kotlinx.android.synthetic.main.card_receivables.view.btn_receivables_history

class StudentFeesFragment : BaseFragment(), StudentFeesContract.View{

    override val TAG: String = "fragment_student_fees"

    override lateinit var presenter: StudentFeesContract.Presenter

    override val applicationNavigator: ApplicationNavigator by lazy {
        ApplicationNavigator(activity as BaseActivity)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_student_fees, container, false)
        initPresenter()
        initListeners(view)
        return view
    }

    override fun onResume() {
        super.onResume()
        presenter.start()
    }

    private fun initListeners(view: View) {
        view.btn_payments_history.setOnClickListener {
            applicationNavigator.goToListActivity(Constants.ListActivityAdapterType.PAYMENT)
        }
        view.btn_receivables_history.setOnClickListener {
            applicationNavigator.goToListActivity(Constants.ListActivityAdapterType.RECEIVABLE)
        }
    }

    private fun initPresenter() {
        presenter = StudentFeesPresenter(
                Injection.provideStudentDataRepository(mContext as Context),
                this,
                mContext as Context
        )
    }

    override fun setBalance(balance: String?) {
        this.balance.text = balance
    }

    override fun setBalanceReceivables(balanceReceivables: String?) {
        this.balance_receivables.text = balanceReceivables
    }

    override fun setBalancePayments(balancePayments: String?) {
        this.balance_payments.text = balancePayments
    }

    override fun setBalanceWithdrawals(balanceWithdrawals: String?) {
        this.balance_withdrawals.text = balanceWithdrawals
    }

    override fun setPaymentsDate1(paymentsDate1: String?) {
        this.payments_date1.text = paymentsDate1
    }

    override fun setPaymentsDate1Amount(paymentsDate1Amount: String?) {
        this.payments_date1_amount.text = paymentsDate1Amount
    }

    override fun setPaymentsDate2(paymentsDate2: String?) {
        this.payments_date2.text = paymentsDate2
    }

    override fun setPaymentsDate2Amount(paymentsDate2Amount: String?) {
        this.payments_date2_amount.text = paymentsDate2Amount
    }

    override fun setReceivablesItem1(receivablesItem1: String?) {
        this.receivables_item1.text = receivablesItem1
    }

    override fun setReceivablesItem1Date(receivablesItem1Date: String?) {
        this.receivables_item1_date.text = receivablesItem1Date
    }

    override fun setReceivablesItem1Amount(receivablesItem1Amount: String?) {
        this.receivables_item1_amount.text = receivablesItem1Amount
    }

    override fun setReceivablesItem2(receivablesItem2: String?) {
        this.receivables_item2.text = receivablesItem2
    }

    override fun setReceivablesItem2Date(receivablesItem2Date: String?) {
        this.receivables_item2_date.text = receivablesItem2Date
    }

    override fun setReceivablesItem2Amount(receivablesItem2Amount: String?) {
        this.receivables_item2_amount.text = receivablesItem2Amount
    }

    override fun setAccount(account: String?) {
        this.account.text = account
    }

    companion object {

        fun newInstance() = StudentFeesFragment()

    }
}